edx/frontend-app-payment
# refactor: [REV-2388] remove "non-profit" and isPriceMessageExperiment

## Description


We want to make sure learners aren't mislead into thinking edX is a non-profit, now that we are no longer one.

This PR:
* Removes non-profit language from frontend-app-payment.
* Deprecates Optimizely `isPriceMessageExperiment` which contains the word "non-profit" but is no longer in use.

### Screenshots

#### Before
Screenshot of master after manually setting `window.experimentVariables.isPriceMessageExperiment = true` in Chrome DevTools, showing `isPriceMessageExperiment`:
 <img width="1199" alt="item-35-before" src="https://user-images.githubusercontent.com/3790674/143784451-a6394b65-1112-4f53-9a9c-10aedde40d16.png">

#### After
Screenshot of branch after manually setting `window.experimentVariables.isPriceMessageExperiment = true` in Chrome DevTools, showing that `isPriceMessageExperiment` no longer exists:
<img width="1199" alt="item-35-after-devstack" src="https://user-images.githubusercontent.com/3790674/143784484-6d6e7543-d8ab-42f3-afd1-6e97baa19541.png">

## Supporting information

* JIRA: [REV-2388](https://openedx.atlassian.net/browse/REV-2388)
* Research: [Confluence](https://openedx.atlassian.net/wiki/spaces/~931919476/pages/3229155368/REV-2388)

## Testing instructions

To simulate Optimizely experiment, run in Chrome DevTools:

```js
window.experimentVariables = {}
window.experimentVariables.isPriceMessageExperiment = true
```
Perform test purchase:
* after manually setting `window.experimentVariables.isPriceMessageExperiment = true`
* verify cart page does not contain `isPriceMessageExperiment`
* verify payment completes successfully

- [X] In dev
- [ ] In staging
- [ ] In production

## Other Information

Special thanks to @marcotuts, who did pre-work on this in #498.

## Checklist
- [X] Consider PCI compliance impact and whether this PR changes how credit card information is handled.
- [X] Intend to [release and monitor](https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories) this PR promptly after merge.
